src/tike/operators/propagation.py

```diff
@@ -6,11 +6,41 @@
 
 
 class Propagation(Operator):
-    """A base class for Fourier-based free-space propagation."""
+    """A base class for Fourier-based free-space propagation.
+
+    Take an (..., N, N) array and pads the last two dimensions with zeros until
+    it is size (..., M, M). Then apply the Fourier transform to the last two
+    dimensions.
+
+    Attributes
+    ----------
+    probe_shape : int
+        The pixel width and height of the nearplane waves.
+    detector_shape : int
+        The pixel width and height of the farplane waves.
+    nwaves : int
+        The number of waves to propagate from the nearplane to farplane.
```

Comment:
what is the difference between nscan and nwaves?

Reply:
The number of scan positions is not equivalent to the number of propagated waves because of multi-mode probes. 5 scan positions and 5 incoherent modes = 25 propagated waves.